### PR TITLE
chore(CDAP-20329): Add SCM feature flag check in handler

### PR DESCRIPTION
**Pull request by de-lan**
### What 
Add SCM feature flag check in handler.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java [hunk 1]
```diff
@@ -46,48 +51,50 @@
...
     RepositoryConfigRequest repoRequest = getRepositoryConfigRequest(request);
 
-    if (repoRequest == null || repoRequest.getConfig() == null) {
-      responder.sendJson(HttpResponseStatus.BAD_REQUEST,
-                         GSON.toJson(new SetRepositoryResponse("Repository configuration must be specified.")));
-    }
-
     try {
+      if (repoRequest == null || repoRequest.getConfig() == null) {
+        throw new InvalidRepositoryConfigException("Repository configuration must be specified.");
```
Comment (de-lan) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java:75]: Moving these lines into the try block and throw `InvalidRepositoryConfigException` for code consistency, also avoid missing returns after the responder sends back

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java [hunk 2]
```diff
@@ -140,6 +160,23 @@ public void testInvalidRepoConfig() throws Exception  {
...
+    RepositoryConfig namespaceRepo = new RepositoryConfig.Builder().setProvider(Provider.GITHUB)
+      .setLink(LINK).setDefaultBranch(DEFAULT_BRANCH).setAuthType(AuthType.PAT)
+      .setTokenName(TOKEN_NAME).setUsername(USERNAME).build();
+
+    // Assert the namespace does not exist
+    assertResponseCode(403, setRepository(NAME, createRepoRequestString(namespaceRepo)));
+    assertResponseCode(403, getRepository(NAME));
+    assertResponseCode(403, deleteRepository(NAME));
+
+    setSCMFeatureFlag(true);
```
Comment (sanjanasandeep) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java]: better to set this in an after/ clean up class method. What if this test fails? then we might not reach this line

Comment (de-lan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java]: nice catch! will add it

Comment (de-lan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java]: actually we just need to remove this line, I'll have `setSCMFeatureFlag(true);` in `@Before`

File: cdap-common/src/main/resources/cdap-default.xml [hunk 3]
```diff
@@ -5506,7 +5506,6 @@
   <property>
     <name>feature.source.control.management.git.enabled</name>
     <value>false</value>
-    <final>true</final>
```
Comment (de-lan) [inline cdap-common/src/main/resources/cdap-default.xml:5509]: This shall not be set to `final` yet

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java [hunk 4]
```diff
@@ -48,6 +55,26 @@ public class SourceControlManagementHttpHandlerTests extends AppFabricTestBase {
...
+
+  @Before
+  public void before() {
+    setSCMFeatureFlag(true);
+  }
+
+  @After
+  public void cleanup() {
+    setSCMFeatureFlag(false);
+  }
```
Comment (arjan-bal) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java]: If we're setting the flag as true before every test, why is this required? The only time the feature flag would be false is b/w two tests. I would also recommend putting `cConf = createBasicCConf();` inside the `@Before` instead of `@BeforeClass` so that cConf changes from one test don't effect another.

**Review by arjan-bal — COMMENTED** [review]
Just a nit, otherwise looks good.

**Review by sanjanasandeep — APPROVED** [review]

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java (hunk 4 above)
Comment (de-lan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java]: sounds good, I've removed this `@After` cleanup. But looks like we cannot `createBasicCConf();` each time for a single case, the config data directory exists once created. Trying to create it again will fail

**Review by albertshau — APPROVED** [review]
